gtda/curves/_functions.py

```diff
@@ -0,0 +1,52 @@
+# License: GNU AGPLv3
+
+import warnings
+from itertools import product
+
+import numpy as np
+from joblib import Parallel, delayed
+
+_AVAILABLE_FUNCTIONS = {
+    "identity": {},
+    "argmax": {},
+    "argmin": {},
+    "min": {},
+    "max": {},
+    "mean": {},
+    "std": {},
+    "median": {},
+    "average": {"weights": {"type": np.ndarray}}
+    }
+
+_implemented_function_recipes = {
+    "identity": lambda X, axis: X.reshape(len(X), -1),
+    "argmax": np.argmax,
+    "argmin": np.argmin,
+    "min": np.min,
+    "max": np.max,
+    "mean": np.mean,
+    "std": np.std,
+    "median": np.median,
+    "average": np.average
+    }
+
+
+def _parallel_featurization(Xt, function, function_params, n_jobs):
+    if callable(function):
+        return function(Xt, axis=-1, **function_params)
```

Comment:
I am not a big fan of having an edge case for the use of parallelism. Even if it is documented, it's likely that I would never read the `n_jobs` part of the doc and I doubt our users will. 

That being said, I think it's not such a big deal. xD

Reply:
For what it's worth, I've added an explicit note in n_jobs saying that it is ignored when one of the strings is passed.

Reply:
I will be compromising here, although parallel on sliced arrays could also have been used so that `n_jobs` is always consistently used xD

Reply:
Haha, ok.  Just academically though, I had a different impression on this issue.  If the user writes down an arbitrary function which works on 1d arrays but will break on 3d or 2d arrays, then slicing isn't an option right?

Reply:
So basically my point is that the slicing logic is compatible with the predefined functions because they are numpy array functions, but not really compatible with arbitrary user-defined functions which satisfy the documentation requirements of mapping 1d -> 1d or scalar.